api/src/core/adapters/hal/getHalSoftwareExternalData.ts

```diff
@@ -28,6 +29,69 @@ const buildParentOrganizationTree = async (
     );
 };
 
+const parseReferencePublication = (key: string, value: string | string[]): ScholarlyArticle[] => {
+    const arrayValue = typeof value === "string" ? value.split(",") : value;
+    switch (key) {
+        case "hal":
+            return arrayValue.map((halThing): ScholarlyArticle => {
```

Comment:
idem, proposition d'explicité ce qui vient de codeMeta :
```suggestion
            return arrayValue.map((halThing): CodeMeta.ScholarlyArticle => {
```